docs/examples/git-clone-go-build.md

```diff
@@ -0,0 +1,132 @@
+---
+layout: default
+title: Git Clone and Go Build
+nav_order: 14
+parent: Examples
+---
+
+# Introduction
+Today we are going to create workflow that can git clone, go build and then upload the new build to Amazon S3. The following YAML snippet is the start of the workflow pulling each container that is required.
+
+```yaml
+id: build-go-binary
+functions:
+- id: go
+  image: vorteil/go:v1
+  files:
+  - key: helloworld
+    scope: instance
+    type: tar.gz
+- id: git
+  image: vorteil/git:v1
+- id: upload
+  image: vorteil/amazon-upload:v3
+  files:
+  - key: helloworldserver
+    scope: instance
+description: "Clones a repository and builds a Go server."
+states:
+  # continued in the next code block
+```
+
+The 'git' container will clone the entire repository and save it as an instance variable called 'helloworld'. Which gets referenced from the 'go' container as you can see via the 'files' inclusion. The 'go' container then uses that variable to build the binary from the repository and upload to the helloworldserver variable. Then the 'upload' container will then upload to S3 on Amazon.
+
+## Git Clone
+I created a simple repository that serves hello-world which can be found [here](https://github.com/vorteil/helloworld). We will be cloning this into an instance variable for the later containers to access.
```

Comment:
For the purposes of this demonstration, I've created a [git repository](https://github.com/vorteil/helloworld) that provides the code required to build the Go binary. The contents of the repository will be saved to an instance variable, to be accessed by subsequent functions/states.